src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -1692,25 +1772,21 @@ impl EventProcessor {
 
     /// Send modifiers for the active window.
     ///
-    /// The event won't be sent when the `modifiers` match the previously `sent` modifiers value.
+    /// The event won't be sent when the `modifiers` match the previously `sent` modifiers value,
+    /// unless `force` is passed. The `force` should be passed when the active window changes.
     fn send_modifiers<T: 'static, F: FnMut(&RootELW, Event<T>)>(
         &self,
+        window_id: crate::window::WindowId,
         modifiers: ModifiersState,
+        force: bool,
         callback: &mut F,
     ) {
-        let window_id = match self.active_window {
-            Some(window) => mkwid(window),
-            None => return,
-        };
-
-        if self.modifiers.replace(modifiers) != modifiers {
-            callback(
-                &self.target,
-                Event::WindowEvent {
-                    window_id,
-                    event: WindowEvent::ModifiersChanged(self.modifiers.get().into()),
-                },
-            );
+        if self.modifiers.replace(modifiers) != modifiers || force {
```

Comment:
Nit: Put `force` first to help with branch prediction.

Reply:
No, it'll be wrong that way since `force and mods != old_mods` won't update the modifiers, maybe I should write it in a comment to make it more clear?

Reply:
Updated with the comment explicitly pointing that out.